src/coupled_driver.cpp

```diff
@@ -578,4 +600,34 @@ void CoupledDriver::set_density()
   }
 }
 
+void CoupledDriver::comm_report()
+{
+  char c[_POSIX_HOST_NAME_MAX];
+  gethostname(c, _POSIX_HOST_NAME_MAX);
```

Comment:
This is a weird one.  MacOS and several other distros in the BSD family don't provide `HOST_NAME_MAX`.  Here's a comment from [syslimits.h in FreeBSD](https://github.com/freebsd/freebsd/blob/7ab2b4d26558ad8bccfdb89538ed1e2f5599ca69/sys/sys/syslimits.h#L72-L80):

```c
/*
 * We leave the following values undefined to force applications to either
 * assume conservative values or call sysconf() to get the current value.
 *
 * HOST_NAME_MAX
 *
 * (We should do this for most of the values currently defined here,
 * but many programs are not prepared to deal with this yet.)
 */
```

The POSIX standard allows for this.  In the [specs for limits.h](https://pubs.opengroup.org/onlinepubs/009695399/basedefs/limits.h.html), `HOST_NAME_MAX` belongs to the category "Runtime Invariant Values (Possibly Indeterminate)":

> A definition of one of the symbolic names in the following list shall be omitted from <limits.h> on specific implementations where the corresponding value is equal to or greater than the stated minimum, but is unspecified.
> 
> This indetermination might depend on the amount of available memory space on a specific instance of a specific implementation. The actual value supported by a specific instance shall be provided by the sysconf() function.

And `_POSIX_HOST_NAME_MAX` belongs to the category "Minimum Values":

> The symbolic constants in the following list shall be defined in <limits.h> with the values shown. These are symbolic names for the most restrictive value for certain features on an implementation conforming to this volume of IEEE Std 1003.1-2001. Related symbolic constants are defined elsewhere in this volume of IEEE Std 1003.1-2001 which reflect the actual implementation and which need not be as restrictive. A conforming implementation shall provide values at least this large. A strictly conforming application must not require a larger value for correct operation.

So `_POSIX_HOST_NAME_MAX` isn't strictly correct, since a distro could have implemented `gethostname` such that it uses `HOST_NAME_MAX` and such that `HOST_NAME_MAX` is larger than `_POSIX_HOST_NAME_MAX`.  

One workaround could look like this:

```c
#if !defined(HOST_NAME_MAX)
#if defined(_POSIX_HOST_NAME_MAX)
#define HOST_NAME_MAX _POSIX_HOST_NAME_MAX
#else 
#define HOST_NAME_MAX 255
#endif 
#endif
```

It's also possible that CMake could discover it using sysconf, but that's just speculation on my part.

Reply:
Strange, let's just stick with `_POSIX_HOST_NAME_MAX`